common/src/WooCommerce.php

```diff
@@ -204,4 +230,124 @@ private function call_method($obj, $name) {
+            $products[] = [
+                'id' => $product->get_id(),
+                'name' => $product->get_name(),
+                'url' => get_permalink($product->get_id()),
+                'image_url' => $this->getProductImage($product),
+                'sku' => $product->get_sku(),
+                'gtin' => $gtin_handler->getGtin(),
+                'reviews_allowed' => $product->get_reviews_allowed(),
+            ];
+        }
+        return $products;
+    }
+
+    private function getProductImage(WC_Product $product) {
+        foreach (get_attached_media('image', $product->get_id()) as $image) {
+            return wp_get_attachment_image_src($image->ID, 'full')[0] ?? null;
+        }
+    }
+
+    public function syncReviews(): void {
+        if (!get_option($this->plugin->getOptionName('product_reviews'))) {
+            return;
+        }
+        $api_domain = $this->plugin->getDashboardDomain();
+        $shop_id = get_option($this->plugin->getOptionName('wwk_shop_id'));
+        $api_key = get_option($this->plugin->getOptionName('wwk_api_key'));
+        $api = new API($api_domain, $shop_id, $api_key);
+
+        $api = new API($api_domain, $shop_id, $api_key);
+        try {
+            $reviews = $api->getReviews();
+        } catch (WebwinkelKeurAPIError $e) {
+            $this->logApiError($e);
+            return;
+        }
+        $this->processReviews($reviews);
+    }
+
+    private function processReviews(\SimpleXMLElement $reviews): void {
+        foreach ($reviews as $review) {
+            $comment_data = $this->getCommentData($review);
+            if (empty($comment_data)) {
+                continue;
+            }
+            $comment_id = $this->getExistingComment(
+                $comment_data['comment_post_ID'],
+                $comment_data['comment_author_email'],
+                (int) $review->review_id
+            );
+            if ($comment_id) {
+                $comment_data['comment_ID'] = $comment_id;
+                wp_update_comment($comment_data);
+            } else {
+                $comment_id = wp_insert_comment($comment_data);
+                update_comment_meta(
+                    $comment_id,
+                    "_{$this->plugin->getOptionName('review_id')}",
+                    (int) $review->review_id
+                );
+            }
+        }
+    }
+
+    private function getExistingComment(int $post_id, string $author_email, int $review_id): ?int {
+        $args = [
+            'post_id' => $post_id,
+            'author_email' => $author_email,
+            'type' => 'review',
+            'meta_query' => [
+                'key' => "_{$this->plugin->getOptionName('review_id')}",
+                'value' => $review_id,
+            ]
+        ];
+        $comments_query = new WP_Comment_Query($args);
+        $comments = $comments_query->comments;
+        return (count($comments) > 0) ? $comments[0]->comment_ID : null;
```

Comment:
`return $comments[0]->comment_ID ?? null;` would work as well and be shorter.  (If you had `?? null` before and I didn't see it, sorry.)